examples/js/loaders/GLTFLoader.js

```diff
@@ -1454,6 +1514,112 @@ THREE.GLTFLoader = ( function () {
 
 	};
 
+	/**
+	 * @param {string} key
+	 * @param {GLTF.definition} def
+	 * @return {Promise<GLTF.definition>}}
+	 */
+	GLTFParser.prototype._onBefore = function ( key, def ) {
+
+		var parser = this;
+		var functionName = 'onBefore' + key[ 0 ].toUpperCase() + key.slice( 1 );
+		var plugins = this.plugins || {};
+		var extensionPlugins = plugins.extensions || {};
+		var extensions = def.extensions || {};
+		var pending = Promise.resolve( def );
+
+		for ( var extensionName in extensions ) {
+
+			var plugin = extensionPlugins[ extensionName ];
+
+			if ( plugin === undefined || plugin[ functionName ] === undefined ) {
+
+				continue;
+
+			}
+
+			pending = pending.then( function ( plugin, def ) {
+
+				return plugin[ functionName ]( def, parser );
```

Comment:
I think we may also want to pass the index of the definition. For example if you have a `onBeforeNode` hook and you want to look up some associated data by node index, you currently don't have that index. Maybe you want to look up all of the animation samplers that target this node, you would need the index of the node to do that.

Reply:
Added, thanks.